docs/practices/workflows/otel_traces.md

```diff
@@ -0,0 +1,135 @@
+# Working with OpenTelemetry Traces
+
+`neard` is instrumented in a few different ways. From the code perspective we have two major ways
+of instrumenting code:
+
+* Prometheus metrics – by computing various metrics in code and expoding them via the `prometheus`
+  crate.
+* Execution tracing – this shows up in the code as invocations of functionality provided by the
+  `tracing` crate.
+
+The focus of this document is to provide information on how to effectively work with the data
+collected by the execution tracing approach to instrumentation.
+
+## Gathering and Viewing the Traces
+
+Tracing the execution of the code produces two distinct types of data: spans and events. These then
+are exposed as either logs (representing mostly the events) seen in the standard output of the
+`neard` process or sent onwards to an [opentelemetry collector].
+
+[opentelemetry collector]: https://opentelemetry.io/docs/collector/
+
+When deciding how to instrument a specific part of the code, consider the following decision tree:
+
+1. Do I need execution timing information? If so, use a span; otherwise
+2. Do I need call stack information? If so, use spans; otherwise
```

Comment:
```suggestion
2. Do I need call stack information? If so, use a span; otherwise
```

For consistency. Or you are saying that multiple spans has to be set up?